repos/system_upgrade/el8toel9/actors/rocescanner/tests/unit_test_rocescanner.py

```diff
@@ -0,0 +1,155 @@
+import pytest
+
+from leapp.libraries.actor import rocescanner
+from leapp.libraries.common.config import architecture
+from leapp.libraries.common.testutils import CurrentActorMocked, logger_mocked, produce_mocked
+from leapp.libraries.stdlib import CalledProcessError
+
+NMCLI_CON_NIC1 = [
+    'ens1: connected to ens1',
+    '"Mellanox MT27710"',
+    'ethernet (mlx5_core), 82:28:9B:1B:28:2C, hw, mtu 1500',
+    'inet4 192.168.0.1/16',
+    'route4 192.168.0.1/16',
+    'inet6 fe80::d8c5:3a67:1abb:dcca/64',
+    'route6 fe80::/64',
+    ''
+]
+
+NMCLI_CON_NIC2 = [
+    'eno2: connected to eno2',
+    '"mellanox MT27710"',
+    'ethernet (mlx5_core), 82:28:9B:1B:28:2C, hw, mtu 1500',
+    'inet4 192.168.0.1/16',
+    'route4 192.168.0.1/16',
+    'inet6 fe80::d8c5:3a67:1abb:dcca/64',
+    'route6 fe80::/64',
+    ''
+]
+
+NMCLI_DISCON_NIC3 = [
+    'ens3: disconnected',
+    '"Mellanox MT27710"',
+    'ethernet (mlx5_core), 82:28:9B:1B:28:2C, hw, mtu 1500',
+    'inet4 192.168.0.1/16',
+    'route4 192.168.0.1/16',
+    'inet6 fe80::d8c5:3a67:1abb:dcca/64',
+    'route6 fe80::/64',
+    ''
```

Comment:
@tdi2004  most likely I put here a nonsense, as disconnected device does not have IP. We should replace it by a real output.

Reply:
yes, indeed. But this will not be the last change, please let me do some real tests tomorrow.
But the commit message is crisp and with good technical content!!